ext/date/php_date.c

```diff
@@ -5260,7 +5260,7 @@ static int php_date_period_initialize_from_hash(php_period_obj *period_obj, Hash
 
 	ht_entry = zend_hash_str_find(myht, "start", sizeof("start")-1);
 	if (ht_entry) {
-		if (Z_TYPE_P(ht_entry) == IS_OBJECT && Z_OBJCE_P(ht_entry) == date_ce_date) {
+		if (Z_TYPE_P(ht_entry) == IS_OBJECT && (Z_OBJCE_P(ht_entry) == date_ce_date || Z_OBJCE_P(ht_entry) == date_ce_immutable)) {
```

Comment:
Should these be equalities or instanceof? I.e., is it possibly for a subclass of DateTime to appear here?

Reply:
Well, actually it's about `instanceof DateTime`; would `instanceof_function(Z_OBJCE_P(ht_entry), date_ce_date` be the correct check?

Reply:
`instanceof_function(Z_OBJCE_P(ht_entry), date_ce_interface)` to cover both.

Reply:
Oops, of course. Thanks!